// Copyright 2017 The Forseti Security Authors. All rights reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package notifier;

service Notifier {
  rpc Ping(PingRequest) returns (PingReply) {}

  rpc Run(RunRequest) returns (stream Progress) {}

}

message RunRequest {
  int64 inventory_index_id = 1;
}

message Progress {
  string server_message = 1;
}

message PingRequest {
  string data = 1;
}

message PingReply {
  string data = 1;
}
